.github/workflows/e2e-nvidia-a10g-x1.yml

```diff
@@ -116,18 +117,41 @@ jobs:
 
       - name: Install ilab
         run: |
-          export PATH="/home/ec2-user/.local/bin:/usr/local/cuda/bin:$PATH"
-          python3.11 -m venv --upgrade-deps venv
+          export CUDA_HOME="/usr/local/cuda"
+          export LD_LIBRRY_PATH="$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/usr/local/cuda/extras/CUPTI/lib64"
+          export PATH="$PATH:$CUDA_HOME/bin"
+          python3.11 -m venv venv
           . venv/bin/activate
+          nvidia-smi
           sed 's/\[.*\]//' requirements.txt > constraints.txt
           python3.11 -m pip cache remove llama_cpp_python
           CMAKE_ARGS="-DLLAMA_CUBLAS=on" python3.11 -m pip install --force-reinstall --no-binary llama_cpp_python -c constraints.txt llama_cpp_python
           python3.11 -m pip install bitsandbytes
+
+          # TODO This should be added to instructlab-training
+          python3.11 -m pip install packaging wheel
+
+          python3.11 -m pip install instructlab-training[cuda]
+
           python3.11 -m pip install .
 
       - name: Run e2e test
         run: |
+          nvidia-smi
+          export PYTORCH_CUDA_ALLOC_CONF=garbage_collection_threshold:0.6,max_split_size_mb:128
           . venv/bin/activate
+          # TODO: for some reason we need to reinstall DS in order to get fused adam support
+          # This means we need to manually rm and re-install a bunch of packages. Investigate why this is.
+          python3.11 -m pip uninstall -y deepspeed
```

Comment:
Is there an open issue tracking this?

Reply:
this is a training repo issue probably. @Maxusmusti WDYT?

Reply:
this was because we were installing deepspeed from a prebuilt binary when we were training on vela: 
https://github.com/instructlab/openshift-manager/blob/cb775725aefeaa1d88f986151e2ca8a1503e3848/ocj2/templates/new_vela_deepspeed_train.yml.j2#L158

I'm not sure if that's still relevant

Reply:
so you're suggesting this could be removed from here?

Reply:
@cdoern the following lines are unecessary now:
```
          nvidia-smi
          python3.11 -m pip show nvidia-nccl-cu12
```
Additionally, the original reason for the reinstall of deepspeed was to have it build the DeepSpeedCPUAdam optimizer at install time, rather than runtime (since there were missing cuda bits on the box for runtime compilation). Since we have resolved the original issues though, there's a chance that this re-install step is removable now, but needs to be tested

Reply:
I'm OK with either resolving this here, or filing an issue for the remaining cleanup related to deepspeed install/uninstall

You could  use a single issue for a list of training-lib-in-e2e cleanup tasks if you want

Reply:
https://github.com/instructlab/instructlab/issues/1655